### PR TITLE
docs: add critical updates from recent development experience

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
- Add critical worktree deletion prevention guidelines to CLAUDE.md
- Document comprehensive GitHub sub-issue management workflow
- Add test coverage for parseTimestampBound function

### CLAUDE.md Updates
- Added reference to phantom worktree management section in issue-management.md
- Emphasized critical rule about never deleting worktrees without explicit permission
- Reduced duplicate content by consolidating with existing documentation

### dev-docs/issue-management.md Updates
- Added comprehensive GitHub native sub-issue management documentation
- Included complete GraphQL workflow with examples
- Provided batch creation script for efficiency
- Documented sub-issue management operations (remove, reorder, replace parent)

### Test Coverage Improvement
- Added comprehensive test for `parseTimestampBound` function in system_variables_test.go
- Covers all 5 timestamp bound types (STRONG, MIN_READ_TIMESTAMP, READ_TIMESTAMP, MAX_STALENESS, EXACT_STALENESS)
- Includes error cases, edge cases, and case insensitivity tests
- Contributes to overall test coverage improvement (#320)

## Context
These updates capture important learnings from recent development sessions:
1. Preventing unauthorized worktree deletions (from issue #277 incident)
2. Proper workflow for creating GitHub native sub-issues (from issue #248 work)
3. Improving test coverage as part of the testing initiative

## Test Plan
- [ ] Verify CLAUDE.md updates are clear and actionable
- [ ] Confirm issue-management.md sub-issue workflow is complete
- [ ] Run tests to ensure parseTimestampBound tests pass: `go test -run TestParseTimestampBound`

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -100,6 +100,7 @@ gh pr comment <PR-number> --body "Development insights..."
 **IMPORTANT**: When working with phantom worktrees, **ALWAYS refer to** [dev-docs/issue-management.md#phantom-worktree-management](dev-docs/issue-management.md#phantom-worktree-management) for detailed guidelines including:
 - AI assistant decision matrix for autonomous vs. user-permission operations
 - Comprehensive deletion safety rules
+- **CRITICAL**: Never delete worktrees without explicit user permission
```
[the remark this answers is not in the document]
Comment (apstndb) [reply CLAUDE.md]: Enhanced the critical rule with a warning symbol (⚠️), made it a separate section titled 'CRITICAL RULE - WORKTREE DELETION', and added detailed explanation of consequences including permanent loss of uncommitted work.

Fixed in commit 4ace631.
